### PR TITLE
Port ODBC adapter to collector #19. Add class diagram.

### PR DESCRIPTION
**Pull request by tigra**
Port ODBC adapter to collector (issue #19). Had to switch to python:3.9.12-slim-buster to be able to install pyodbc.
Also added PlantUML class diagram for Adapter hierarchy (pydantic domain objects).

### CODE REVIEW

File: Dockerfile [hunk 1]
```diff
@@ -11,20 +12,40 @@ RUN apt-get update && \
...
 
+# For pyodbc
+COPY --from=build microsoft.asc microsoft.asc
+COPY --from=build mssql-release.list mssql-release.list
+
+ENV ACCEPT_EULA=Y
+RUN apt-get update -y && apt-get install -y gnupg2 \
+    && apt-key add microsoft.asc && rm microsoft.asc && \
+    mv mssql-release.list /etc/apt/sources.list.d/mssql-release.list
+RUN apt-get update -y && apt-get install -y unixodbc-bin
```
Comment (Vixtir) [inline Dockerfile:41]: Do we really need to create new layer for each install command and call `apt-get update` every time?

File: Dockerfile [hunk 2]
```diff
@@ -11,20 +12,40 @@ RUN apt-get update && \
...
+RUN apt-get update -y && apt-get install -y msodbcsql17
+RUN apt-get install -y libgssapi-krb5-2
+
+
 RUN useradd --create-home --shell /bin/bash app
-RUN apt-get update && \
-    apt-get install -y libpq-dev
+#RUN apt-get update && \
```
Comment (Vixtir) [inline Dockerfile:47]: Remove comments

File: odd_collector/adapters/odbc/adapter.py [hunk 3]
```diff
@@ -0,0 +1,89 @@
+import logging
+
+import pyodbc
```
Comment (Vixtir) [inline odd_collector/adapters/odbc/adapter.py:3]: Adapter module shouldn't know about external library as pyodbc.

File: odd_collector/adapters/odbc/adapter.py [hunk 4]
```diff
@@ -0,0 +1,89 @@
...
+
+                entities = map_table(self.__oddrn_generator, tables, columns)
+                return entities
+        except Exception as e:
+            logging.error("Failed to load metadata for tables")
+            logging.exception(e)
+
+            return []
+
+    # replace
```
Comment (Vixtir) [inline odd_collector/adapters/odbc/adapter.py:67]: replace?

**Review by Vixtir — CHANGES_REQUESTED** [review]

File: odd_collector/adapters/odbc/adapter.py (hunk 3 above)
Comment (tigra) [reply odd_collector/adapters/odbc/adapter.py]: Decoupled repository from adapter.

File: odd_collector/adapters/odbc/adapter.py (hunk 4 above)
Comment (tigra) [reply odd_collector/adapters/odbc/adapter.py]: Got rid of this

**Review by Vixtir — APPROVED** [review]

**Pull request merged**